### PR TITLE
feature: radio button as a widget

### PR DESCRIPTION
**Pull request by arpitnath**
**Fixes**: #483 

** Radio button as a widget**
Screenshot 2021-08-12 at 4 16 43 PM

Screenshot 2021-08-12 at 10 52 00 AM

**Description** : 
   component: RadioButton
   events: onSelectionChange
   properties: label, value, values, display_values
   default widget size: width: 150px, height: 50px

**Loom**: 
https://www.loom.com/share/c7235278a41344808e0aca0c6c40e7da

### CODE REVIEW

File: frontend/src/Editor/Components/RadioButton.jsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    selectOptions = [
+      ...parsedValues.map((value, index) => {
+        return { name: parsedDisplayValues[index], value: value };
+      })
+    ];
+  } catch (err) { console.log(err); }
+
+
+  function onSelect(event) {
+    const checked = event.target.value
```
Comment (Navaneeth-pk) [inline frontend/src/Editor/Components/RadioButton.jsx:47]: will `const selection` be a better name for the variable?

File: frontend/src/Editor/Components/RadioButton.jsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    onComponentOptionChanged(component, 'value', checked);
+    if (checked) {
+      onEvent('onCheck', { component });
```
Comment (Navaneeth-pk) [inline frontend/src/Editor/Components/RadioButton.jsx:50]: `onCheck` event works better for checkboxes, here the event should be `onSelectionChange`.

File: frontend/src/Editor/Components/components.js [hunk 3]
```diff
@@ -365,6 +365,56 @@ export const componentTypes = [
...
+      showOnMobile: { type: 'toggle', displayName: 'Show on mobile?'},
+    },
+    properties: {
+      label: { type: 'code', displayName: 'Label' },
+      value: { type: 'code', displayName: 'Default value' },
+      values: { type: 'code', displayName: 'Option values' },
+      display_values: { type: 'code', displayName: 'Option labels' }
+    },
+    events: {
+      onCheck: { displayName: 'On select'},
```
Comment (Navaneeth-pk) [inline frontend/src/Editor/Components/components.js:388]: event should be `onSelectionChange`

File: frontend/src/Editor/Components/RadioButton.jsx (hunk 2 above)
Comment (arpitnath) [reply frontend/src/Editor/Components/RadioButton.jsx:50]: okay 👍

File: frontend/src/Editor/Components/components.js [hunk 4]
```diff
@@ -365,6 +365,56 @@ export const componentTypes = [
...
+    definition: {
+      others: {
+        showOnDesktop: { value: true },
+        showOnMobile: { value : false }
+      },
+      properties: {
+        label: { value: 'Select' },
+        value: { value: '' },
+        values: { value: '{{[1,2,3]}}' },
+        display_values: { value: '{{["one", "two", "three"]}}' },
```
Comment (Navaneeth-pk) [inline frontend/src/Editor/Components/components.js:403]: Does `yes` and `no` sounds more related to radio buttons?

Comment (Navaneeth-pk) [reply frontend/src/Editor/Components/components.js:403]: `yes` and `no` with `true` and `false` as values.

Comment (arpitnath) [reply frontend/src/Editor/Components/components.js:403]: yeah or `on` or `off`?

Comment (arpitnath) [reply frontend/src/Editor/Components/components.js:403]: for sample values, I think `yes` or `no` will be sound better

**Comment by Navaneeth-pk** [comment]
@arpitnath From the screenshot it looks like we should reduce the default width of the widget.

File: frontend/src/Editor/Components/components.js [hunk 5]
```diff
@@ -365,6 +365,56 @@ export const componentTypes = [
...
+    exposedVariables: {},
+    definition: {
+      others: {
+        showOnDesktop: { value: true },
+        showOnMobile: { value : false }
+      },
+      properties: {
+        label: { value: 'Select' },
+        value: { value: '' },
+        values: { value: '{{[1,2]}}' },
```
[the remark this answers is not in the document]
Comment (arpitnath) [reply frontend/src/Editor/Components/components.js:402]: yeah I have, just need to push

**Comment by arpitnath** [comment]
Updated the values to [true, false] and updated the size. Also updated the screenshots and loom recording

**Comment by Navaneeth-pk** [comment]
image

The label and the options should be aligned.

**Comment by arpitnath** [comment]
Screenshot 2021-08-13 at 10 50 27 AM

Updated the changes.
`width: 200px`

**Pull request merged**